packages/snaps-jest/src/internals/request.ts

```diff
@@ -101,12 +103,24 @@ export function handleRequest({
     }) as unknown as SnapRequest;
 
   promise.getInterface = async () => {
-    return await runSaga(
+    const sagaPromise = runSaga(
       getInterface,
       runSaga,
       snapId,
       controllerMessenger,
     ).toPromise();
+    const result = await Promise.race([promise, sagaPromise]);
+
+    // If the request promise has resolved to an error, we should throw
+    // instead of waiting for an interface that likely will never be displayed
+    if (
+      is(result, SnapResponseStruct) &&
+      hasProperty(result.response, 'error')
+    ) {
+      throw result.response.error;
```

Comment:
@Mrtenz Any thoughts on what error message we should use here?

Reply:
Ideally we could give them a better error message that explains why 🤔

Reply:
Yeah, true. We could throw "The Snap returned an invalid interface." and revisit when we have a better error message?

Reply:
Sounds good!

Reply:
Actually, this may be thrown on other types of errors too 🤔 

Since it just should throw if the RPC request fails for whatever reason.

So we cannot assume it to be because the interface is invalid

Reply:
> Unable to get the interface from the Snap: The returned interface may be invalid. The error message received was: "error here".

Reply:
That works 👍